### ISSUE TITLE
Error plotting density2d

### ISSUE THREAD

**Issue by JS3xton** [taborlab/FlowCal#249]
I get an error on a particular FCS file when I try to generate a 2D density plot.

```python
import FlowCal as fc
import matplotlib.pyplot as plt
d = fc.io.FCSData('density_plot_fail.fcs')
fc.plot.density2d(d, channels=['FSC-A', 'SSC-A'])
plt.show()
```

yields

```
Traceback (most recent call last):
  File "C:\Users\sexto\Anaconda2\envs\FlowCalv1.1.2\lib\site-packages\matplotlib\backends\backend_qt5agg.py", line 197, in __draw_idle_agg
    FigureCanvasAgg.draw(self)
  File "C:\Users\sexto\Anaconda2\envs\FlowCalv1.1.2\lib\site-packages\matplotlib\backends\backend_agg.py", line 464, in draw
    self.figure.draw(self.renderer)
  File "C:\Users\sexto\Anaconda2\envs\FlowCalv1.1.2\lib\site-packages\matplotlib\artist.py", line 63, in draw_wrapper
    draw(artist, renderer, *args, **kwargs)
  File "C:\Users\sexto\Anaconda2\envs\FlowCalv1.1.2\lib\site-packages\matplotlib\figure.py", line 1143, in draw
    renderer, self, dsu, self.suppressComposite)
  File "C:\Users\sexto\Anaconda2\envs\FlowCalv1.1.2\lib\site-packages\matplotlib\image.py", line 139, in _draw_list_compositing_images
    a.draw(renderer)
  File "C:\Users\sexto\Anaconda2\envs\FlowCalv1.1.2\lib\site-packages\matplotlib\artist.py", line 63, in draw_wrapper
    draw(artist, renderer, *args, **kwargs)
  File "C:\Users\sexto\Anaconda2\envs\FlowCalv1.1.2\lib\site-packages\matplotlib\axes\_base.py", line 2409, in draw
    mimage._draw_list_compositing_images(renderer, self, dsu)
  File "C:\Users\sexto\Anaconda2\envs\FlowCalv1.1.2\lib\site-packages\matplotlib\image.py", line 139, in _draw_list_compositing_images
    a.draw(renderer)
  File "C:\Users\sexto\Anaconda2\envs\FlowCalv1.1.2\lib\site-packages\matplotlib\artist.py", line 63, in draw_wrapper
    draw(artist, renderer, *args, **kwargs)
  File "C:\Users\sexto\Anaconda2\envs\FlowCalv1.1.2\lib\site-packages\matplotlib\axis.py", line 1136, in draw
    ticks_to_draw = self._update_ticks(renderer)
  File "C:\Users\sexto\Anaconda2\envs\FlowCalv1.1.2\lib\site-packages\matplotlib\axis.py", line 969, in _update_ticks
    tick_tups = [t for t in self.iter_ticks()]
  File "C:\Users\sexto\Anaconda2\envs\FlowCalv1.1.2\lib\site-packages\matplotlib\axis.py", line 912, in iter_ticks
    majorLocs = self.major.locator()
  File "build\bdist.win-amd64\egg\FlowCal\plot.py", line 429, in __call__
    return self.tick_values(vmin, vmax)
  File "build\bdist.win-amd64\egg\FlowCal\plot.py", line 482, in tick_values
    total_ticks = log_range[1] - log_range[0]
UnboundLocalError: local variable 'log_range' referenced before assignment
```

[density_plot_fail.zip](https://github.com/taborlab/FlowCal/files/956504/density_plot_fail.zip)

**Comment by JS3xton** [comment]
Another file that looks like it fails in the same way:

[density_plot_fail2.zip](https://github.com/taborlab/FlowCal/files/982471/density_plot_fail2.zip)

**Comment by castillohair** [comment]
The issue goes deeper than I thought. The code is not handling properly the presence of the "linear" portion of the logicle scale when the "logarithmic" portion is not present. Even zooming on the "linear" region of the logicle plot can result in an error with files that otherwise work.

For example, running the following code against file ``Data002.fcs`` from the example files:

```python
import FlowCal as fc
import matplotlib.pyplot as plt

d = fc.io.FCSData('Data002.fcs')
d = fc.transform.to_rfi(d)
fc.plot.density2d(d, channels=['FSC', 'SSC'], mode='scatter')
plt.xlim(0, 2)
plt.show()
```

Gives an error with the current develop (ab708b08095a712c3ad39e32f636fb7bfe963fde):
```
Traceback (most recent call last):
  File "C:\Users\casti\Anaconda2\lib\site-packages\matplotlib\backends\backend_qt5agg.py", line 197, in __draw_idle_agg
    FigureCanvasAgg.draw(self)
  File "C:\Users\casti\Anaconda2\lib\site-packages\matplotlib\backends\backend_agg.py", line 464, in draw
    self.figure.draw(self.renderer)
  File "C:\Users\casti\Anaconda2\lib\site-packages\matplotlib\artist.py", line 63, in draw_wrapper
    draw(artist, renderer, *args, **kwargs)
  File "C:\Users\casti\Anaconda2\lib\site-packages\matplotlib\figure.py", line 1144, in draw
    renderer, self, dsu, self.suppressComposite)
  File "C:\Users\casti\Anaconda2\lib\site-packages\matplotlib\image.py", line 139, in _draw_list_compositing_images
    a.draw(renderer)
  File "C:\Users\casti\Anaconda2\lib\site-packages\matplotlib\artist.py", line 63, in draw_wrapper
    draw(artist, renderer, *args, **kwargs)
  File "C:\Users\casti\Anaconda2\lib\site-packages\matplotlib\axes\_base.py", line 2426, in draw
    mimage._draw_list_compositing_images(renderer, self, dsu)
  File "C:\Users\casti\Anaconda2\lib\site-packages\matplotlib\image.py", line 139, in _draw_list_compositing_images
    a.draw(renderer)
  File "C:\Users\casti\Anaconda2\lib\site-packages\matplotlib\artist.py", line 63, in draw_wrapper
    draw(artist, renderer, *args, **kwargs)
  File "C:\Users\casti\Anaconda2\lib\site-packages\matplotlib\axis.py", line 1136, in draw
    ticks_to_draw = self._update_ticks(renderer)
  File "C:\Users\casti\Anaconda2\lib\site-packages\matplotlib\axis.py", line 969, in _update_ticks
    tick_tups = [t for t in self.iter_ticks()]
  File "C:\Users\casti\Anaconda2\lib\site-packages\matplotlib\axis.py", line 918, in iter_ticks
    minorLocs = self.minor.locator()
  File "d:\castillohair\documents\rice\lab\programs\flowcal\FlowCal\plot.py", line 429, in __call__
    return self.tick_values(vmin, vmax)
  File "d:\castillohair\documents\rice\lab\programs\flowcal\FlowCal\plot.py", line 509, in tick_values
    decade_next_low = min(decades[np.nonzero(decades)]) / b
ValueError: min() arg is an empty sequence
```

I have a tentative fix (805d53e2cde263b520dade4f0ee894db54fd65c0) that has, so far, undergone limited testing. The same code above produces the following:
![image](https://user-images.githubusercontent.com/1529481/27164524-7faf80f4-5153-11e7-8cb4-4c4acd35fe7c.png)

Which is what I would expect. The two files that @JS3xton submitted also work now. I will test it a lot more before submitting a pull request.
